sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs

```diff
@@ -99,6 +99,48 @@ public class EventData
         ///
         public string PartitionKey { get; }
 
+        /// <summary>
+        ///   The sequence number that was last enqueued into the Event Hub partition from which this
+        ///   event was received.
+        /// </summary>
+        ///
+        /// <remarks>
+        ///   This property is only populated for events received from the Event Hubs service and when
+        ///   partition metrics have been requested.
+        /// </remarks>
+        ///
+        /// <seealso cref="EventHubConsumerOptions.RequestPartitionMetricUpdates" />
+        ///
+        public long? LastPartitionSequenceNumber { get; }
+
+        /// <summary>
+        ///   The offset that was last enqueued into the Event Hub partition from which this event was
+        ///   received.
+        /// </summary>
+        ///
+        /// <remarks>
+        ///   This property is only populated for events received from the Event Hubs service and when
+        ///   partition metrics have been requested.
+        /// </remarks>
+        ///
+        /// <seealso cref="EventHubConsumerOptions.RequestPartitionMetricUpdates" />
+        ///
+        public long? LastPartitionOffset { get; }
+
+        /// <summary>
+        ///   The date and time, in UTC, that the last event was enqueued into the Event Hub partition from
+        ///   which this event was received.
+        /// </summary>
+        ///
+        /// <remarks>
+        ///   This property is only populated for events received from the Event Hubs service and when
+        ///   partition metrics have been requested.
+        /// </remarks>
+        ///
+        /// <seealso cref="EventHubConsumerOptions.RequestPartitionMetricUpdates" />
+        ///
+        public DateTimeOffset? LastPartitionEnqueuedTime { get; }
```

Comment:
Can we have the `LastPartitionSequenceNumber`, `LastPartitionOffset` and `LastPartitionEnqueuedTime` as internal?
Given that this information will get exposed on the consumer, is there is need to make these first class citizens and publicly accessible to the user from `EventData`?

Reply:
This was done for three reasons:  

1) allowing developers to mock  (could be offset by `protected internal` but cannot be only `internal`)
2) conforming to the track one pattern where it is exposed
3) you asked to expose it on the event for preview 3 in our conversation

Happy to move it to `protected internal` if you'd like.

Reply:
> you asked to expose it on the event for preview 3 in our conversation

That was assuming that we are not exposing the same on the consumer. I wanted to give an out if exposing these on the consumer was not trivial. It can stay as is for Preview 3, not a big deal

Reply:
So long as it doesn't break a bunch of tests _(it shouldn't)_  I'll mark it.   All good.